libs/components/modals/src/lib/modules/modal/modal-is-dirty.directive.ts

```diff
@@ -0,0 +1,99 @@
+import { Directive, Input, OnDestroy, OnInit } from '@angular/core';
+import { SkyLibResourcesService } from '@skyux/i18n';
+
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+import { SkyConfirmType } from '../confirm/confirm-type';
+import { SkyConfirmService } from '../confirm/confirm.service';
+
+import { SkyModalBeforeCloseHandler } from './modal-before-close-handler';
+import { SkyModalInstance } from './modal-instance';
+
+/**
+ * Provides support for flagging a modal as dirty which will
+ * automatically prompt the user whether or not to discard changes
+ * when the user tries closing a dirty modal without
+ * saving or canceling the modal.
+ */
+@Directive({
+  // Since this is limited to sky-modal, it should be safe to
+  // leave off the sky prefix.
+  // eslint-disable-next-line @angular-eslint/directive-selector
+  selector: 'sky-modal[isDirty]',
+})
+export class SkyModalIsDirtyDirective implements OnInit, OnDestroy {
+  /**
+   * Flags whether or not the modal is considered dirty.
+   * @required
+   */
+  @Input()
+  public isDirty = false;
+
+  readonly #ngUnsubscribe = new Subject<void>();
+
+  readonly #modalInstance: SkyModalInstance;
```

Comment:
Sure, I can make these changes. I wasn't sure if that's the way you all were leaning because there benefits to injecting all dependencies into constructors. Java devs used `@Autowire` for a long time until they determined it was bad practice because it didn't allow variables to be marked as final (not an issue here), and it also made things harder to unit test because you had to fully buy into a Dependency Injection system since that was the only way to easily set those private fields. Now, it is best practice to inject all dependencies into the constructor in Java.

Reply:
I'm not worried about that, especially for components, because we would never construct them outside Angular's `TestBed`. For services you can provide the service in `TestBed` along with mocks (where needed) for injected tokens.